packages/phrases-experience/src/types.ts

```diff
@@ -1,10 +1,17 @@
 import type en from './locales/en/index.js';
 
-type FlattenKeys<T, Prefix extends string = ''> = {
-  [K in keyof T]: T[K] extends Record<string, unknown>
-    ? `${Prefix}${string & K}.${FlattenKeys<T[K]>}`
-    : `${Prefix}${string & K}`;
-}[keyof T];
+type FlattenKeys<
+  T,
+  Prefix extends string = '',
+  D extends number = 11, // Depth limit is actually 10, since the initial value of A is [0]
```

Comment:
i doubt if it's sustainable. which code will cause the error?

Reply:
I think maybe a potential `typescript` version bump